ChangeLog.d/fix-compilation-when-memcpy-is-function-like-macro.txt

```diff
@@ -0,0 +1,2 @@
+Bugfix
+   * Fix compilation error when memcpy() is a function-like macros. Fixes #8994.
```

Comment:
Unfortunately this might come back by accident, possibly on a different standard library function. (With internal functions, it's ok, we know that they aren't macros.) It's not really practical to test for this in the CI (I don't think this unusual platform justifies a new test component).

Maybe we can at least document it? `MBEDTLS_SSL_TLS1_3_LBL_WITH_LEN` doesn't have documentation at all. It would be nice to have some usage guidance.

(And more generally, I'm not fond of unhygienic macros like `MBEDTLS_SSL_TLS1_3_LBL_WITH_LEN`, but in this case I think it's a lesser evil compared with the risk of mismatching pointer and size, which has a very high security impact.)

Reply:
> this might come back by accident

Actually, I disagree - I think this type of macro is very unusual, and matching it up to a C library call even more unusual, so I don't think we lose anything by addressing any such failures if and when they arise.

> Maybe we can at least document it?

Yes, it needs to be documented, but that is clearly out of scope of this PR. I'll raise an issue if you like